### PR TITLE
Set upper version limit for geopandas until #829 is solved

### PR DESCRIPTION
**Pull request by ClaraBuettner**
Deals with #829 

## Before merging into `dev`-branch, please make sure that

- [x] the `CHANGELOG.rst` was updated.
- [ ] new and adjusted code is formated using `black` and `isort`.
- [ ] the `Dataset`-version is updated when existing datasets are adjusted.
- [x] the branch was merged into the
      `continuous-integration/run-everything-over-the-weekend`-branch.
- [x] the workflow is running successful in `test mode`.
- [x] the workflow is running successful in `Everything` mode.

### CODE REVIEW

**Review by nesnoj — APPROVED** [review]
As we don't have a stable `dev` due to this problem (I just ran into this again), it'd be good to have this fixed.
From my point of view this can be merged straightforward as all of us have `0.10.2` installed anyway.

Roll back as soon as the tasks in #829 are solved.

**Comment by ClaraBuettner** [comment]
I think we can merge this into `dev`, I was just waiting for a review from @gnn.

**Pull request merged**
